tutorials/28_grant_posting_permission/README.md

```diff
@@ -0,0 +1,159 @@
+
+The tutorial is set up with three individual functions for each of the required operations - checking permission status, granting permission and revoking permission.
+
+## Steps
+
+1.  [**Configure connection**](#connection) Configuration of `dsteem` to communicate with a Steem blockchain
+2.  [**Input variables**](#input) Collecting the required inputs via an HTML UI.
+3.  [**Database query**](#query) Sending a query to the blockchain for the posting permissions (status)
+4.  [**Object creation**](#object) Create the array and subsequent data object for the broadcast operation
+5.  [**Broadcast operation**](#broadcast) Broadcasting the changes to the blockchain
+
+#### 1. Configure connection<a name="connection"></a>
+
+As usual, we have a `public/app.js` file which holds the Javascript segment of the tutorial. In the first few lines we define the configured library and packages:
+
+```javascript
+const dsteem = require('dsteem');
+//define network parameters
+let opts = {};
+opts.addressPrefix = 'STX';
+opts.chainId =
+    '79276aea5d4877d9a25892eaa01b0adf019d3e5cb12a97478df3298ccdd01673';
+//connect to a steem node, testnet in this case
+const client = new dsteem.Client('https://testnet.steem.vc', opts);
+```
+
+Above, we have `dsteem` pointing to the testnet with the proper chainId, addressPrefix, and endpoint. Due to this tutorial altering the blockchain it is preferable to not work on production.
+
+#### 2. Input variables<a name="input"></a>
+
+The required parameters for the account status query is recorded via an HTML UI that can be found in the `public/index.html` file. The values are pre-populated in this case but any account name can be used.
+
+All of the functions use the same input variables. Once the function is activated via the UI the variables are allocated as seen below.
+
+```javascript
+    //get username
+    const username = document.getElementById('username').value;
+    //get private active key
+    const privateKey = dsteem.PrivateKey.fromString(
+        document.getElementById('privateKey').value
+    );
+    //get account to provide posting auth
+    const newAccount = document.getElementById('newAccount').value;
+```
+
+#### 3. Database query<a name="query"></a>
+
+The queries are sent through to the steem blockchain with the `database API` using the `getAccounts` function. The results of the query is used to check the status of the current posting authorisations and parameters as per the `intro`.
+
+```javascript
+    //query database for posting array
+    _data = new Array
+    _data = await client.database.getAccounts([username]);
+    const postingAuth = _data[0].posting;
+
+    //check for username duplication
+    const checkAuth = _data[0].posting.account_auths;
+    var arrayindex = -1;
+    for (var i = 0,len = checkAuth.length; i<len; i++) {
+        if (checkAuth[i][0]==newAccount) {
+            arrayindex = i
+            var checktext = "User already has posting permission"
```

Comment:
For readability, it would be better if these messages were templates, with the usernames injected, so that the messages are clear. The messages themselves will need to change of course.